extensions/ql-vscode/src/vscode-tests/cli-integration/helpers.test.ts

```diff
@@ -0,0 +1,39 @@
+import * as path from 'path';
+import { extensions } from 'vscode';
+import 'mocha';
+
+import { CodeQLCliServer } from '../../cli';
+import { CodeQLExtensionInterface } from '../../extension';
+import { tryGetQueryMetadata } from '../../helpers';
+import { expect } from 'chai';
+
+describe('helpers (with CLI)', function() {
+  const baseDir = path.join(__dirname, '../../../src/vscode-tests/cli-integration');
+
+  // up to 3 minutes per test
+  this.timeout(3 * 60 * 1000);
+
+  let cli: CodeQLCliServer;
+
+  beforeEach(async () => {
+    const extension = await extensions.getExtension<CodeQLExtensionInterface | Record<string, never>>('GitHub.vscode-codeql')!.activate();
+    if ('cliServer' in extension) {
+      cli = extension.cliServer;
+    } else {
+      throw new Error('Extension not initialized. Make sure cli is downloaded and installed properly.');
+    }
+  });
+
+  it('should get query metadata if it exists', async () => {
```

Comment:
Extremely minor nit: split this into two test functions with different labels, one for no metadata and one for metadata. Then you get distinct test failures for each case.

Reply:
Good idea, thanks! Done in the latest commit 😺